azkaban-common/src/main/java/azkaban/executor/OnContainerizedExecutionEventListener.java

```diff
@@ -70,22 +70,44 @@ private void restartExecutableFlow(final ExecutableFlow exFlow) {
     // Update the flow options so that the flow will be not retried again by Azkaban
 
     // inherent the retry time counters
-    executableFlow.setUserDefinedRetryCount(exFlow.getUserDefinedRetryCount());
-    executableFlow.setSystemDefinedRetryCount(exFlow.getSystemDefinedRetryCount());
+    retryExFlow.setUserDefinedRetryCount(originalExFlow.getUserDefinedRetryCount());
+    retryExFlow.setSystemDefinedRetryCount(originalExFlow.getSystemDefinedRetryCount());
+
+    if (originalExFlow.getFlowRetryRootExecutionID() > 0) {
```

Comment:
We should set those under if they're "allow.restart" flag is enabled. 
They could be null, as they're not enabled;
They could be 0, as retry not restarted yet.
I feel like current definition (0,-1,null) is a bit confusing to me,

Reply:
we will only enter this method when auto-retry is defined;
plus this value is default to -1 at initialization, so no null concern